pynicotine/core.py

```diff
@@ -333,6 +333,7 @@ def _server_disconnect(self, manual_disconnect=False):
         self.user_status = slskmessages.UserStatus.OFFLINE
 
         # Clean up connections
+        self.user_addresses.clear()
```

Comment:
Crash on quit, after interacting with a private chat of a non-existant user who is never online:

```
Error in sys.excepthook:
Traceback (most recent call last):
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/mainwindow.py", line 362, in on_window_active_changed
    self.privatechat.clear_notifications()
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/privatechat.py", line 133, in clear_notifications
    self.window.application.notifications.clear("private", user=user)
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/widgets/notifications.py", line 71, in clear
    if core.notifications.remove_hilite_item(location, item):
  File "/home/user/Git/slook/nicotine-plus/pynicotine/notifications.py", line 57, in remove_hilite_item
    if item not in self.chat_hilites[location]:
KeyError: 'private'

Original exception was:
Traceback (most recent call last):
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/mainwindow.py", line 362, in on_window_active_changed
    self.privatechat.clear_notifications()
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/privatechat.py", line 133, in clear_notifications
    self.window.application.notifications.clear("private", user=user)
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/widgets/notifications.py", line 71, in clear
    if core.notifications.remove_hilite_item(location, item):
  File "/home/user/Git/slook/nicotine-plus/pynicotine/notifications.py", line 57, in remove_hilite_item
    if item not in self.chat_hilites[location]:
KeyError: 'private'
/usr/lib/python3/dist-packages/gi/overrides/Gio.py:43: Warning: Source ID 9479 was not found when attempting to remove it
  return Gio.Application.run(self, *args, **kwargs)
```

Reply:
That's a different issue.

Reply:
Moved to discussion https://github.com/nicotine-plus/nicotine-plus/discussions/2145#discussioncomment-4783141